@navikt/core/react/src/util/composition-warning/CompositionWarning.tsx

```diff
@@ -0,0 +1,77 @@
+/**
+ * Give warnings based on component composition (which slot/parent a child is rendered in).
+ *
+ * Used when child components need to know which slot/parent they are rendered in
+ * (e.g. `FormSummary.Header` vs `FormSummary.Footer`) and should warn or error in development
+ * if placed in a discouraged or forbidden slot.
+ *
+ * Usage:
+ * - Wrap slot components with <CompositionWarning.Root name="FormSummary.Header">...</CompositionWarning.Root>
+ * - In child: `<CompositionWarningForbidden.Forbidden name="FormSummary.Header" />` to forbid slot.
+ *
+ * This is guidance only: warnings are logged to the console in development, never enforced at runtime.
+ */
+import React, { useEffect, useRef } from "react";
+import { Slot } from "../../slot/Slot";
+import { createContext } from "../create-context";
+
+type CompositionName = string;
+
+/**
+ * Allows developers to manually enable usage warnings in any browser environment
+ * (excluding production) by setting `window.__AKSEL_DEBUG_WARNINGS__ = true`
+ * in the browser console. Useful for local debugging when NODE_ENV is not "development".
+ */
+const forcedDebug =
+  typeof window !== "undefined" &&
+  (window as any).__AKSEL_DEBUG_WARNINGS__ === true &&
+  process.env.NODE_ENV !== "production";
+
+/**
+ * Enables warnings only in development or when forcedDebug is set.
+ */
+const isDev = forcedDebug || process.env.NODE_ENV === "development";
+
+type CompositionWarningContextType = {
+  name: CompositionName;
+};
+
+const [CompositionWarning, useCompositionWarning] =
+  createContext<CompositionWarningContextType>({
+    errorMessage:
+      "useCompositionWarning() must be used within <CompositionWarning />",
+  });
+
+type CompositionWarningForbiddenProps = {
+  children?: React.ReactElement;
+  /**
+   * Name of the composition that is forbidden.
```

Comment:
I find this description confusing. What about something like this?

```suggestion
   * Name of the parent slot component where the child is not allowed.
```